wolfcrypt/src/asn.c

```diff
@@ -136,6 +136,7 @@ extern int wc_InitRsaHw(RsaKey* key);
 #endif
 
 #if defined(OPENSSL_EXTRA) || defined(OPENSSL_EXTRA_X509_SMALL)
+    #include <wolfssl/internal.h>
```

Comment:
Can we avoid including `wolfssl/internal.h`? This seems problematic going forward and breaks the wolfssl/wolfcrypt separation. If there are things asn.c needs they should be in asn.h.

Reply:
`wolfssl/openssl/objects.h` is already included on the next line which in turn includes `wolfssl/ssl.h`. Removing the second include results in a long list of missing symbols meaning that `asn.c` is already very closely tied to wolfSSL. I don't think adding this include changes much in the wolfssl/wolfcrypt separation. Someone could in the future investigate the feasibility to refactor out the wolfssl parts back into the `src` folder.